api/src/main/java/com/codeforcommunity/rest/ApiRouter.java

```diff
@@ -564,6 +576,46 @@ private void handleGetEventUsers(RoutingContext ctx) {
     }
     response.end(userJson);
   }
+  
+  private void handleResetPasswordEmail (RoutingContext ctx) {
+    HttpServerResponse response = ctx.response();
+    HttpServerRequest request = ctx.request();
+    String email = request.params().get("email"); 
+    
+    if(!email.endsWith("@husky.neu.edu")) {
+      response.setStatusCode(400).setStatusMessage("Incorrect email. Must be of type @husky.neu.edu.").end();
+    }
+    
+    //if email is valid send email
+    if(processor.validateEmail(email)) {
+      String token = createJWT(email, "auth-token", TOKEN_DURATION * 24, 0, true);
+      resetTokenMap.put(email, token);
+      EmailUtil.sendEmail(email, token);
+    }
+    else response.setStatusCode(400).setStatusMessage("Invalid email.").end();
+    
+    response.setStatusCode(200).end();
+    
+        
+  }
+  
+  private void handleResetPassword (RoutingContext ctx) {
+    HttpServerResponse response = ctx.response();
+    HttpServerRequest request = ctx.request();
+    String qs = request.params().get("qs");
+    String email = request.params().get("email");
+    String password = request.params().get("password"); 
+    
+    if(resetTokenMap.get(email).equals(qs)) { //right now is in memory so if server crashes all tokens will be reset
```

Comment:
This doesn't check for the validity of the JWT, and so expired password reset tokens will still go through!